src/autodiff/variables/variable.rs

```diff
@@ -49,8 +49,9 @@ pub struct Variable<'v> {
 
 impl<'v> Variable<'v> {
     /// Instantiate a new variable.
+    #[must_use]
     #[inline]
-    pub fn new(graph: &'v Graph, index: usize, value: f64) -> Self {
+    pub const fn new(graph: &'v Graph, index: usize, value: f64) -> Self {
```

Comment:
What's the reasoning here ?

Reply:
It allows you to create a const variable with the constructor.